stories/topics/proc-aap-gcp-application-upgrade.adoc

```diff
@@ -0,0 +1,131 @@
+[id="proc-aap-gcp-application-upgrade"]
+
+Technical Preview
+~~~~~~~~~~~~~~~~~
+
+You can upgrade your existing {AAPonGCP} deployment to the newer version. The upgrade process covers upgrading of automation hub and automation controller. The upgrade process takes roughly the same amount of time as a {AAPonGCP} deployment install. Follow the steps below to ensure a smooth upgrade process.
+
+[NOTE]
+=====
+{AAPonGCP} supports sequential upgrades. All upgrades should be no more than one major version behind what you are currently upgrading to. For example, in order to upgrade to {AAPonGCP} to 2.2.20230215, you must be on the first version of {AAPonGCP}.
+=====
+
+.Prerequisites
+* You require the docker installed in order to run the upgrade playbook.
+
+== Preparing the environment
+
+The upgrade process requires a few volumes to be mounted, so *prepare a fresh directory* to be used for this process.
+
+.Procedure
+. First, create the directories that will be used to configure the upgrade process.
++
+[source,bash]
+----
+# Secrets directory. Store the GCP credentials file.
+$ mkdir secrets
+# Extra vars directory. All the configuration variables for the upgrade process.
+$ mkdir extra_vars
+----
++
+[NOTE]
+=====
+To create the GCP credentials file, see, 
+link:https://developers.google.com/workspace/guides/create-credentials[Create credentials] and 
+link:https://cloud.google.com/iam/docs/keys-create-delete[Create and delete keys]
+=====
++
+. Store your Service account credential json file in secrets dir. 
++
+. Gather the following input variables to create an `extra_vars.yaml` file within `extra_vars` dir.
+.. *gcp_service_account_credentials_json_path*: Service account credential json file path for the google cloud account.
+.. *gcp_deployment_name*: Deployment name which you want to upgrade.
+.. *gcp_compute_region*: GCP region which you provided when deploying foundation deployment. If you haven't provided region when deploying foundation, use default region here which is `us-east1`.
+.. *gcp_compute_zone*: GCP zone which you provided when deploying foundation deployment. If you haven't provided zone when deploying foundation, use default zone here which is `us-east1-b`.
++
+[NOTE]
+=====
+You can find the region and zone from the filestore instance deployed with the foundation deployment. Look for filestore with name <deployment-name>-filestore and make note of Location field that is your `gcp_compute_zone` and remove the last 2 bits from the location that is your `gcp_compute_region`. For example, filestore instance location is `us-eat1-d`, your `gcp_compute_zone` is `us-east1-d` and your region is `us-east1`.
+=====
++
+[source,bash]
+----
+gcp_deployment_name: "" 
+gcp_compute_region: ""
+gcp_compute_zone: ""
+gcp_service_account_credentials_json_path: "/secrets/<service-account-creds-json-filename>"
```

Comment:
@leena-jawale:

there is still some clarity missing in terms of absolute path for secrets JSON file

The tree command does not specify if it was run from collections repo or elsewhere and extra_vars.yaml specifies `gcp_service_account_credentials_json_path: "/secrets/<service-account-creds-json-filename>"`

which makes me think secrets dir exists in my laptop's root '/' directory


You need to either update the tree command or extra_vars.yaml to show the right absolute path

Reply:
all commands shown in the doc have '$' prompt, hence having the file in `/secrets/<service-account-creds-json-filename>` as shown in extra_vars.yaml is not going to work for regular users